### PR TITLE
feat: add copy success notification for Field API names

### PR DESCRIPTION
**Pull request by dyncan**
## Describe your changes

Whenever a user clicks on an API Name after enabling "Show Fields API Names," a notification will pop up confirming that the field has been successfully copied to the clipboard. This immediate feedback ensures users that the copy action was successful.

feat_copyFieldSuccessNotification

## Issue ticket number and link
N/A
## Checklist before requesting a review
- [x] I have read and understand the [Contributions section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [x] Target branch is releaseCandidate and not master
- [x] I have performed a self-review of my code
- [x] I ran the [unit tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests) and my PR does not break any tests
- [x] I documented the changes I've made on the [CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md) and followed actual conventions
- [ ] I added a new section on [how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md) (optional)

### CODE REVIEW

**Comment by tprouvot** [comment]
Hi @dyncan thanks for this PR !
I understand the goal and think that it may be useful for users however I'm not a big fan of the UI.
The toast is not really aligned with the extension's UI nor Salesforce.

How about something like this ? (same as the toast, it should disappear few seconds later)

image

**Comment by dyncan** [comment]
Hi @tprouvot 

Thank you for your valuable feedback on the PR!

Taking your feedback into account, I've gone ahead and implemented some UI optimizations, I'd love to hear your thoughts on these updates.

Thank you once again for guiding the improvement process.

**Comment by tprouvot** [comment]
Great thank you for this quick update ! 
Would you mind create a feature request explaining the goal of this improvement ? 
Then reference it in the CHANGES.md file like the other ones.

I'm divided about the message, "copied ✓" looks a little bit weird to me but I like to give the indication to users that the action was a copy (many peeps have no idea about the possibility to copy the field name).
On the other hand, showing only "✓" looks better but don't give the info ...

So here's some options:

- "Copied ✓"
- "Copy Success ✓"
- "Successfully copied"
- "✓"

image

**Comment by dyncan** [comment]
Hi @tprouvot, I've created a Feature Request and also updated the CHANGES.md.

Among the options you've provided, I think "Copied ✓" is the most suitable choice. It's concise, clearly communicates the action that has been completed, and the checkmark (✓) provides a universally understood symbol of completion and success. 

Thank you.

File: CHANGES.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - Add performance On data export with option to disable in Options/Data Export tab (contribution by [Nathan Shulman](https://github.com/nshulman))
 - Clean up popup header and footer (contribution by [Nathan Shulman](https://github.com/nshulman))
 - Fix double "Show all data" button [issue 63](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/63) (contribution by [Luca Bassani](https://github.com/baslu93))
+- Enhanced the user interface with a subtle inline "copied" indicator for field copy actions [feature 351](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/351)
```
Comment (tprouvot) [inline CHANGES.md:22]:
```suggestion
- Enhanced the user interface with a subtle inline "copied" indicator for field copy actions [feature 351](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/351) (contribution by [dyncan](https://github.com/dyncan))

```

File: addon/button.js [hunk 2]
```diff
@@ -171,8 +171,37 @@ function initButton(sfHost, inInspector) {
...
-      navigator.clipboard.writeText(e.target.innerText);
+    // Function to handle copy action
+    function copy(e) {
+      // Retrieve the text content of the target element triggered by the event
+      const originalText = e.target.innerText; // Save the original text
+      // Attempt to copy the text to the clipboard
+      navigator.clipboard.writeText(originalText).then(() => {
+        // Create a new span element to show the copy success indicator
+        const copiedIndicator = document.createElement("span");
+        copiedIndicator.textContent = "copied ✓"; // Set the text content to indicate success
```
Comment (tprouvot) [inline addon/button.js:182]:
```suggestion
        copiedIndicator.textContent = "Copied ✓"; // Set the text content to indicate success
```

**Comment by tprouvot** [comment]
Last thing, could you revert the changes applied to package.lock ?

**Comment by dyncan** [comment]
Hi@tprouvot,  Reverted the package-lock.json file.

Thanks!

**Pull request merged**